v3/services/zip_file_generator.rb

```diff
@@ -0,0 +1,53 @@
+require 'zip'
+
+class ZipFileGenerator
+  # Zip the input directory.
+  def self.write(input_dir, output_file)
+    new(input_dir, output_file)
+  end
+
+  attr_accessor :input_dir, :output_file
+  def initialize(input_dir, output_file)
```

Comment:
Let's call these `src` and `dst`, since those are common terms and very easily recognizable.